build.gradle

```diff
@@ -15,7 +15,7 @@ repositories {
 
 dependencies {
     implementation 'org.slf4j:slf4j-api:1.7.9'
-
+    compile files('3rd/gnu-crypto.jar')
```

Comment:
adding jars in the project is a bad practice, you can replace this with `implementation 'org.gnu:gnu-crypto:2.0.1'` since the artifact is on Maven Central

Reply:
the password was added into the FileVolume.so it can be used in all Volumes.
I do this to minimize code changes。